transformer_engine/jax/cpp_extensions.py

```diff
@@ -2662,6 +2664,445 @@ def cross_fused_attn_bwd(q: jnp.ndarray, kv: jnp.ndarray, bias: jnp.ndarray,
                                                            is_training=is_training)
 
 
+class FusedAttnFwdPrimitive(BasePrimitive):
```

Comment:
I see lot of boilerplate replication between `SelfFusedAttn`, `CrossFusedAttn` and `FusedAttn` primitives.

We should consider implementing a base primitive class like `BaseFusedAttn` and then specialize into the `QKVPackedFusedAttn`, `KVPackedFusedAttn`, `UnpackedFusedAttn` subtypes (similar to how `Softmax` primitive specializes into `Scaled`, `ScaledMasked` and `ScaledUpperTriangMasked`) to make this more maintainable.